src/lib.rs

```diff
@@ -210,17 +218,24 @@ impl Tokenizer {
         out.reserve(text.as_bytes().len());
         let words = self.word_split.find_iter(&text).map(|m| m.as_str());
         for word in words {
-            let start_index = out.len();
-            out.extend(
-                word.as_bytes()
-                    .iter()
-                    .map(|b| self.byte_to_token[usize::from(*b)]),
-            );
-            if start_index < out.len() {
-                // If we added anything, mark last character as end-of-word token
-                out.last_mut().unwrap().0 += 256;
+            if word == "<start_of_text>" {
+                out.push(self.start_of_text());
+            } else if word == "<end_of_text>" {
+                out.push(self.end_of_text());
+            } else {
```

Comment:
I would probably still go with the continue on the earlier branches (instead of else) to reduce rightward drift in this block (which represents the common case).